src/sections/create-dataset/DatasetForm.tsx

```diff
@@ -39,14 +40,28 @@ export const DatasetForm = ({
     collectionId,
     onCreateDatasetError
   )
+  const { user } = useSession()
 
   const isErrorLoadingMetadataBlocks = Boolean(errorLoadingMetadataBlocks)
 
   const form = useForm({
     mode: 'onChange',
     defaultValues: formDefaultValues
   })
-
+  const { setValue, reset } = form
+  useEffect(() => {
+    if (user) {
+      setValue('citation.author.0.authorName', user.displayName)
+      setValue('citation.datasetContact.0.datasetContactName', user.displayName)
+      setValue('citation.datasetContact.0.datasetContactEmail', user.email, {
+        shouldValidate: true
+      })
+      if (user.affiliation) {
+        setValue('citation.datasetContact.0.datasetContactAffiliation', user.affiliation)
+        setValue('citation.author.0.authorAffiliation', user.affiliation)
+      }
+    }
+  }, [form, user])
```

Comment:
You don't need `form` inside the dependencies. Also eslint is warning that you need to include `setValue`.

`}, [setValue, user])`